### PR TITLE
Port rewrite and rewrite_js tests into elixir

### PR DESCRIPTION
**Pull request by jjrodrig**
## Overview

This PR ports these tests to Elixir
 - rewrite.js
 - rewrite_js.js

 rewrite.js was partially ported before this PR, now it is completely ported. 

 Rawresp module has been moved and now can be used by any test if a non-parsed http response body is required.

## Testing recommendations

```
 make elixir
```
## Related Issues or Pull Requests
N/A

## Checklist

- [X] Code is written and works correctly
- [X] Changes are covered by tests
- [ ] Any new configurable parameters are documented in `rel/overlay/etc/default.ini`
- [ ] A PR for documentation changes has been made in https://github.com/apache/couchdb-documentation

### CODE REVIEW

File: test/elixir/lib/couch.ex [hunk 1]
```diff
@@ -175,3 +175,109 @@ defmodule Couch do
     end
   end
 end
+
+defmodule Rawresp do
```
Comment (dottorblaster) [inline test/elixir/lib/couch.ex:179]: couldn't we just write a new file for this? Too much pain? (at the moment I can't remember if we could call those functions inside the module this way)

**Review by dottorblaster — CHANGES_REQUESTED** [review]
Astonishing 🐱 

Could you just look into that comment?

File: test/elixir/lib/couch.ex (hunk 1 above)
Comment (jjrodrig) [reply test/elixir/lib/couch.ex:179]: Sure, i'll put it in a new file

Really, a tried to do it with the current Couch module at couch.ex but I don't know how to parametrize this behaviour.

Comment (jjrodrig) [reply test/elixir/lib/couch.ex]: Done

**Review by dottorblaster — APPROVED** [review]
LGTM

**Comment by jjrodrig** [comment]
Thanks!

**Pull request merged**